File: backend/WebApi/WebApi/Hub/ChatMessageDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApi.HubSpace
{
    public class ChatMessageDto
    {

        public long UserId { get; set; }
        public string Message { get; set; }
        public long TargetUserId { get; set; }
        public int? Side { get; set; }
        public int? ReadState { get; set; }
        public DateTime? CreationTime { get; set; }
    }
}

File: frontend/web-client/src/model/citizen/ChatMessageDto.ts
import { FriendshipState } from "src/environments/constant";
import { NumberLiteralType } from "typescript";

export class Message {
    userId?:number;
    key?:string;
    value?:Array<string>;
}
export class ChatMessageDto {
    userId?: number;
    userConnectionId?: string;
    message?: string;
    targetUserId?: number;
    targetConnectionId?: string;
    side?: number;
    readState?: number;
    creationTime?: Date;
}
export class FriendShipDto {
    friendShipId?:number;
    userId?:number;
    friendUserId?: number;
    friendUserName?: string;
    friendProfilePictureId?: string;
    isOnline?: boolean;
    state?: FriendshipState;
}
export class UserNotFriend{
    userId?:number;
    userName?:string;
    roleId?:number;
    urlImage?:string;
    state?: FriendshipState = 2;
}